p2p/connection.go

```diff
@@ -460,30 +479,63 @@ func (c *Connection) processReceives() {
 	for ConnectionOnline == c.state {
 		var message Parcel
 		verbose(c.peer.PeerIdent(), "Connection.processReceives() called. State: %s", c.ConnectionState())
+
+		var bufReader = bufio.NewReaderSize(c.conn, NetworkReadBufferSize)
+
+		b, _ := bufReader.Peek(2048)
```

Comment:
Yes, this is only temporary, so we take a look at what's causing the decoder to crash.